### PR TITLE
Add API docs

### PR DESCRIPTION
**Pull request by tomglenn**
Closes https://github.com/heroiclabs/docs-centre/issues/239

Once merged GH Pages will be deployed for `docs/` folder.

@lugehorsam we'll wait for your review as we've removed references to lombak.var for javadocs generation. See https://github.com/heroiclabs/nakama-java/pull/60/files#diff-35a3ac4af683271940fa8429eca8afb409d03c3708f3fe0efa9141178964c378

image

### CODE REVIEW

File: src/main/java/com/heroiclabs/nakama/DefaultClient.java [hunk 1]
```diff
@@ -30,11 +30,11 @@
 import io.grpc.stub.MetadataUtils;
 import lombok.NonNull;
 import lombok.extern.slf4j.Slf4j;
-import lombok.var;
```
Comment (seanpackham) [inline src/main/java/com/heroiclabs/nakama/DefaultClient.java:33]: @lugehorsam here is where we edited a source file to generate the docs. Could you check if this is ok?

**Comment by seanpackham** [comment]
@mofirouz who would be best to review this before the end of the year (if possible)?

File: README.md [hunk 2]
```diff
@@ -202,6 +202,20 @@ Jitpack makes builds of each commit on its own servers. You can view the results
...
+## Generate Docs
+
+API docs are generated with javadoc and deployed to GitHub pages.
+
+When changing the API comments, rerun javadoc and commit the changes in `docs/*`.
+
+To run javadoc:
+
+```
+cd nakama-java/src/main/com/heroiclabs/nakama
```
Comment (mofirouz) [inline README.md:215]: this should be `src/main/java/com/heroiclabs/nakama` (notice missing `java`)

Comment (mofirouz) [reply README.md:215]: As a matter of fact - these two lines need to be consolidated into this:

---

In the root of the repo, run the following command:

```
./gradlew javadoc
```

---

**Comment by mofirouz** [comment]
I've made some changes to the Gradle script to make it Lombok aware (and do delomboking properly). It now works as expected and all `@Getter` types are documented properly.

Also the process is made much simpler by using `gradle javadoc` task. The output remains `/docs` as wanted.

Feel free to squash+rebase+merge this.

**Pull request merged**